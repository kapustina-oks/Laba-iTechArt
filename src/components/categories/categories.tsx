import { FC, useEffect, useState } from "react";
import { dataItems, getResource } from "../../services/dataService";
import Card from "@/components/card/card";
import { useParams } from "react-router-dom";

const Categories: FC = (): JSX.Element => {
  const { categories } = useParams();

  const [category, setCategory] = useState<string>("");
  const [gameList, setGameList] = useState<dataItems[]>([]);
  const onCategoryLoaded = (newCategory: string): void => {
    setCategory(newCategory);
  };
  const onRequest = () => {
    getResource(`/api/games?category=${categories}`).then((data) => setGameList(() => data));
  };
  useEffect(() => {
    onCategoryLoaded(categories);
    onRequest();
  }, [categories]);
  const content = <Card games={gameList} />;
  return (
    <>
      <div className="home_container">{content}</div>
      ;
    </>
  );
};

export default Categories;
