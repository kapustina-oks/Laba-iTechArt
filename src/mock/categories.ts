// export const categories = {
//   pc: {
//     id: 1,
//     name: "pc",
//     title: "PC",
//   },
//   playstation: {
//     id: 2,
//     name: "playstation",
//     title: "playstation 5",
//   },
//   xbox: {
//     id: 3,
//     name: "xbox",
//     title: "Xbox",
//   },
// }
