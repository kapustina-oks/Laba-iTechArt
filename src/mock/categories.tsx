import mockServerHelper from "webpack-mock-server/lib/mockServerHelper";

export const categories = {
  pc: {
    id: mockServerHelper.getUniqueIdInt(),
    name: "pc",
    title: "PC",
    icons: "fab fa-windows",
  },
  playstation: {
    id: mockServerHelper.getUniqueIdInt(),
    name: "playstation",
    title: "playstation 5",
    icons: "fab fa-playstation",
  },
  xbox: {
    id: mockServerHelper.getUniqueIdInt(),
    name: "xbox",
    title: "Xbox",
    icons: "fab fa-xbox",
  },
};
