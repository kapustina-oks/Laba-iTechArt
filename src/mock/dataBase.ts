const dataGames = [
  {
    name: "Game of Thrones",
    id: 1,
    img: "https://s2.gaming-cdn.com/images/products/5848/orig/game-steam-a-game-of-thrones-genesis-cover.jpg",
    rating: 4,
    price: "50$",
    categories: ["playstation", "pc", "xbox"],
    date: new Date("2021-11-13T19:17:17.064Z"),
    description:
      '"A Game of Thrones - Genesis" immerses you into the heart of the battles and intrigues between the Houses that shaped the Kingdom of Westeros. From Nymeria\'s arrival in the Kingdom of Dorne to the awakening of the "Others" beyond the Wall, you\'ll live the origins of A Game of Thrones saga through more than 1000 years of history, by taking part in Westeros\' founding events and largest battles. ',
  },
  {
    name: "Cyberpunk",
    id: 2,
    img: "https://www.digiseller.ru/preview/213678/p1_3056404_b1cddb3f.jpg",
    rating: 5,
    price: "60$",
    categories: ["xbox", "playstation"],
    date: new Date("2021-11-16T19:17:17.064Z"),
    description:
      "Cyberpunk 2077 is an open-world, action-adventure story set in Night City, a megalopolis obsessed with power, glamour and body modification. You play as V, a mercenary outlaw going after a one-of-a-kind implant that is the key to immortality.\n",
  },
  {
    name: "World of Tanks",
    id: 3,
    img: "https://static-cdn.jtvnw.net/ttv-boxart/World%20of%20Tanks:%20Blitz.jpg",
    rating: 4,
    price: "45$",
    categories: ["pc"],
    date: new Date("2021-11-15T19:17:17.064Z"),
    description:
      "Jump into the free-to-play team-based shooter with an ever-expanding roster of historical vehicles, stunning graphics, spectacular locales, and orchestral scores. Battle it out in the company of other players in thrilling military conflicts. A unique mix of strategy and action awaits!",
  },
  {
    name: "FarCry 6",
    id: 4,
    img: "https://cdn1.epicgames.com/b4565296c22549e4830c13bc7506642d/offer/TETRA_PREORDER_STANDARD%20EDITION_EPIC_Store_Portrait_1200x1600-1200x1600-ca8b802ff13813c37a44ebf68d0946a2.png",
    rating: 5,
    price: "55$",
    categories: ["xbox", "pc"],
    date: new Date("2021-11-12T19:17:17.064Z"),
    description:
      "Welcome to Yara, a tropical paradise frozen in time. As the dictator of Yara, Antón Castillo is intent on restoring his nation back to its former glory by any means, with his son, Diego, following in his bloody footsteps. Their oppressive rule has ignited a revolution.Dive into the gritty world of a modern-day guerrilla revolution to liberate a nation from its oppressive dictators.",
  },
  {
    name: "Warcraft III: Reign of Chaos",
    id: 5,
    img: "http://s01.riotpixels.net/data/a6/6f/a66fd8bf-9d76-445d-805d-fb941f4dcc61.jpg/cover.warcraft-3-reign-of-chaos.500x703.2002-10-21.386.jpg",
    rating: 4,
    price: "65$",
    categories: ["pc", "xbox"],
    date: new Date("2021-11-11T19:17:17.064Z"),
    description:
      "Choose from four races - The Orcs, humans, undead, and the night elves all have separate, unique campaigns that tie into each other.Multiple Neutral buildings, units and monsters to offer new risks, treasures and challenges.",
  },
  {
    name: "Minecraft",
    id: 6,
    img: "https://upload.wikimedia.org/wikipedia/ru/d/da/Minecraft_Dungeons.jpeg",
    rating: 3,
    price: "70$",
    categories: ["xbox"],
    date: new Date("2021-11-14T19:17:17.064Z"),
    description:
      "Brave the dungeons alone, or team up with friends! Up to four players can battle together through action-packed, treasure-stuffed, wildly varied levels—all in an epic quest to save the villagers and take down the evil Arch-Illager!",
  },
];

export default dataGames;

//Math.random().toFixed(4)
