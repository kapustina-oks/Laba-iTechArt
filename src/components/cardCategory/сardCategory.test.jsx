import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { categories } from "../../mock/categories";
import CardCategory from "./cardCategory";

describe("CardCategory", () => {
  it("renders contain some text", () => {
    render(<CardCategory category={categories.pc} />);
    expect(screen.getByText("PC")).toBeInTheDocument();
  });

  it("CardCategory snapshot", () => {
    render(<CardCategory category={categories.xbox} />);
    expect(document.body).toMatchSnapshot();
  });
});
