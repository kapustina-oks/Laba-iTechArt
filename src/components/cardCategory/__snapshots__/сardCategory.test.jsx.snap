// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCategory CardCategory snapshot 1`] = `
<body>
  <div>
    <div
      class="card-category"
    >
      <div
        class="fab-icons"
      >
        <i
          class="fab fa-xbox"
        />
      </div>
      <div
        class="card-title"
      >
        Xbox
      </div>
    </div>
  </div>
</body>
`;

exports[`CardCategory renders without crashing 1`] = `
<body>
  <div>
    <div
      class="card-category"
    >
      <div
        class="fab-icons"
      >
        <i
          class="fab fa-windows"
        />
      </div>
      <div
        class="card-title"
      >
        PC
      </div>
    </div>
  </div>
</body>
`;
